src/psyclone/domain/lfric/lfric_arg_descriptor.py

```diff
@@ -292,21 +300,25 @@ def _validate_field(self, arg_type):
                 format(arg_type.args[0]))
 
         # There must be at least 3 arguments
-        if len(arg_type.args) < 3:
+        if self._nargs < 3:
             raise ParseError(
                 "In the LFRic API each 'meta_arg' entry must have at "
                 "least 3 arguments if its first argument is of {0} type, "
                 "but found {1} in '{2}'.".
                 format(LFRicArgDescriptor.VALID_FIELD_NAMES,
-                       len(arg_type.args), arg_type))
+                       self._nargs, arg_type))
         # There must be at most 4 arguments
-        if len(arg_type.args) > 4:
+        if self._nargs > 4:
             raise ParseError(
                 "In the LFRic API each 'meta_arg' entry must have at "
                 "most 4 arguments if its first argument is of {0} type, "
                 "but found {1} in '{2}'.".
                 format(LFRicArgDescriptor.VALID_FIELD_NAMES,
-                       len(arg_type.args), arg_type))
+                       self._nargs, arg_type))
+
+        # Field data_type is "real" for now, but will be determined by
```

Comment:
I went for `_validate` originally as there seems to be many `_validate_*` methods in LFRic API that do the mix of checking and property setting. I am happy to rename the methods but it would be good to have some developer guidance when to use `_validate*` and when to use `_init*` .

Reply:
Renamed `_validate_*` to `_init_*` methods as requested.

Reply:
Thanks.